structs.go

```diff
@@ -89,6 +99,16 @@ func (s *Struct) Map() map[string]interface{} {
 			}
 		}
 
+		if tagOpts.Has("string") {
+			_, ok := val.Type().MethodByName("String")
+			if !ok {
+				panic(fmt.Sprintf("field %s is not a stringer", name))
+			}
+
+			out[name] = val.MethodByName("String").Call([]reflect.Value{})[0].String()
+			continue
```

Comment:
* Can you rename from `string` to `stringer`? 
* We should not panic, we should just continue. It should be lazily loaded
* This is a better approach:
```go
if s, ok := val.Interface().(fmt.Stringer); ok {
	out[name] = s.String()
}
```
* Also put this after the last block, so it should come **after** the block:

```go
if IsStruct(val.Interface()) && !tagOpts.Has("omitnested") {
}
else {
} 
...


< Put code here> 
```